lib/postgrex/type_module.ex

```diff
@@ -464,7 +464,7 @@ defmodule Postgrex.TypeModule do
         decode_tuple_dispatch(extension, format, rest, oids, n, acc)
       end
 
-    quote do
+    quote generated: true do
```

Comment:
This covers the code starting at line 500:
```
        case Postgrex.Types.fetch(oid, types) do
          {:ok, {:binary, type}} ->
            unquote(oids) = rem - 1

            case [type | types] do
              unquote(dispatch)
            end
```

That `case [type | types] do` code is tricky - it _looks_ like a boring "prepend to list" operation, but AFAIK `types` is never a list (it's a `Postgres.Types.state`!) so it's always making an improper list.